app/services/hyku_addons/qa_select_service.rb

```diff
@@ -41,8 +42,10 @@ def model_name
       end
 
       def tenant_locale
-        @tenant_locale ||= begin
-          return unless (account = Site.instance&.account)
+        @_tenant_locale ||= begin
+          return @locale.to_s.upcase if @locale.present?
+          account = Site.instance&.account
+          return unless account.present?
```

Comment:
This method is getting a bit hectic. We have two guard statements and an inline OR assignment, so this is looking more and more like a code smell that we can clean this up and make it simpler. 

By adding a helper method for `account` and adding an `&.` we can really clean up the method. Then by moving both of the `.to_s.upcase` calls to the result of the assignment, we have a single line method that seem much easier to reason with. 

```ruby
      def tenant_locale
        @tenant_locale ||= (@locale || account&.settings&.dig("locale_name").presence || account&.name).to_s.upcase
      end

      def account
        Site.instance&.account
      end
```

Reply:
Personally, I dont like all of the `&.`'s. It would be much nicer if `NilSite` returned a `NilAccount` that mimicked the `Account` class, rather than just returning `nil` and causing these kinds of affects.